s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Headers.scala

```diff
@@ -71,30 +72,67 @@ object StorageClass {
 
 /**
  * Documentation: http://docs.aws.amazon.com/AmazonS3/latest/dev/serv-side-encryption.html
- * @param algorithm AES-256 or aws:kms
- * @param kmsKeyId optional amazon resource name in the "arn:aws:kms:my-region:my-account-id:key/my-key-id" format.
- * @param context optional base64-encoded UTF-8 string holding JSON with the encryption context key-value pairs
  */
-sealed abstract class ServerSideEncryption(algorithm: String,
-                                           kmsKeyId: Option[String] = None,
-                                           context: Option[String] = None) {
-  def headers: immutable.Seq[HttpHeader] = algorithm match {
-    case "AES256" => RawHeader("x-amz-server-side-encryption", "AES256") :: Nil
-    case "aws:kms" if kmsKeyId.isDefined && context.isEmpty =>
-      RawHeader("x-amz-server-side-encryption", "aws:kms") ::
-      RawHeader("x-amz-server-side-encryption-aws-kms-key-id", kmsKeyId.get) ::
-      Nil
-    case "aws:kms" if kmsKeyId.isDefined && context.isDefined =>
-      RawHeader("x-amz-server-side-encryption", "aws:kms") ::
-      RawHeader("x-amz-server-side-encryption-aws-kms-key-id", kmsKeyId.get) ::
-      RawHeader("x-amz-server-side-encryption-context", context.get) ::
-      Nil
-    case _ => throw new IllegalArgumentException("Unsupported encryption algorithm.")
-  }
+sealed abstract class ServerSideEncryption {
+  def headers: immutable.Seq[HttpHeader]
+
+  def headers(request: S3Request): immutable.Seq[HttpHeader]
```

Comment:
It is not obvious what this `headers` method is supposed to do. Is  `applicableFor` a better name?

Reply:
Before created this issue, **_ServerSideEncryption_** class had this method:
	_**def headers: immutable.Seq[HttpHeader]**_
and this method return all headers according with the specified server side encryption
but actually this headers are definied by the current request.
Maybe I should change the other method:
	**_def headers(request: S3Request):  immutable.Seq[HttpHeader]_**
by 
	**_def headersByRequest(request: S3Request):  immutable.Seq[HttpHeader]_**

Reply:
Yes, or `headersFor`?